piff/gp_interp.py

```diff
@@ -288,8 +292,11 @@ def interpolateList(self, stars, logger=None):
                 y0_updated = star.fit.get_params(self._num)
             for j in range(self.nparams):
                 y0_updated[self.rows[j]] = y0[j]
-            fit = star.fit.newParams(y0_updated, num=self._num)
-            fitted_stars.append(Star(star.data, fit))
+            if inplace:
+                star.fit.updateParams(y0_updated, num=self._num)
+            else:
+                star = Star(star.data, star.fit.newParams(y0_updated, num=self._num))
+            fitted_stars.append(star)
```

Comment:
I am not as much an expert on the codebase as you are, and this might be totally fine, but I wanted to highlight that the pervious behavior created new starts for the fitted_stars list. Now if this in place they share the same object as the input stars. Could you get action at a distance here where someone updates one of the old references and changes the fitted stars? I figure it is faster to ask, as you probably know the answer vs delaying you by digging into it.